pkg/bpf/detect_linux.go

```diff
@@ -498,7 +498,7 @@ func detectUprobeRefCtrOffset() bool {
 }
 
 func detectUprobeRefCtrOffsetOnce() {
-	batchUpdate.init.Do(func() {
+	uprobeRefCtrOffset.init.Do(func() {
```

Comment:
LGTM, but I wonder: Can we use `OnceFunc`  https://go.dev/pkg/sync/?m=old#OnceFunc for this and other detection functions?